src/bfs.c

```diff
@@ -0,0 +1,49 @@
+#include "bfs.h"
+
+// Function to initialise the attributes of each node
+struct attributes* init_attributes(struct attributes *nodes, int n) {
+	int i;
+	for (i = 1; i <= n; i++) {
+		nodes[i].parent = NIL;
+		nodes[i].distance = INF;
+		nodes[i].current_color = WHITE;
+	}
+}
+
+// Function to print the value of attributes
+void print_attributes(struct attributes *nodes, int n) {
+	int i;
+	printf("Node\tParent\tDistance\n");
+	for (i = 1; i <= n; i++)
+		printf("%4d : %6d\t%7d\n", i, nodes[i].parent, nodes[i].distance);
+}
+
+// Function to implement BFS on a graph
+void bfs(struct Graph *graph, struct attributes *nodes, int source) {
+	int i;
+	nodes = (struct attributes *)malloc((graph->V+1)*sizeof(struct attributes));
+	init_attributes(nodes, graph->V);
+	struct Queue *Q;
+	to_list(graph);
+	nodes[source].current_color = GRAY;
+	nodes[source].distance = 0;
+	Q = init_queue(Q);
+	enqueue(Q, source);
```

Comment:
Keep declarations and assignments separated with at least one line for readability.